pkg/kademlia/kademlia.go

```diff
@@ -54,6 +55,11 @@ type Kademlia struct {
 // NewKademlia returns a newly configured Kademlia instance
 func NewKademlia(id dht.NodeID, bootstrapNodes []pb.Node, address string, identity *provider.FullIdentity) (*Kademlia, error) {
 	self := pb.Node{Id: id.String(), Address: &pb.NodeAddress{Address: address}}
+
+	if _, err := os.Stat("db"); os.IsNotExist(err) {
```

Comment:
Shouldn't this use some form of `filepath.Join(defaultConfDir, "db")` ?

Reply:
I don't really care where it is. If there is a "correct" place I'll put it there. I that where you have typically seen things like this placed ?

Reply:
Talked offline about this:

From @brimstone :
ah, it's 0777, because the umask is applied AFTER you set that
so if you set 0770, then others will never be able to read it, regardless of the umask
so, 0777 means "set all the bits on and let umask sort it out"